### PR TITLE
Add filters coverage in panache-rest module

### PR DESCRIPTION
**Pull request by Sgitario**
The objective of these changes is to coverage the `@Filter` and reproduce the missing functionality to use `@FilterJoinTable` (reported by https://github.com/quarkusio/quarkus/issues/13831).

### CODE REVIEW

File: 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java [hunk 1]
```diff
@@ -76,38 +80,96 @@ public void shouldReturnBadRequestIfApplicationNameIsNull() {
...
+        applicationPath().body(actualApplicationEntity).put("/" + actualApplicationEntity.id)
                 .then().statusCode(HttpStatus.SC_NO_CONTENT);
     }
 
     private void whenDeleteApplication() {
-        applicationPath().delete("/" + actualEntity.id)
+        for (ServiceEntity service : actualServiceEntities) {
+            servicePath().delete("/" + service.id)
+                    .then().statusCode(HttpStatus.SC_NO_CONTENT);
+        }
```
Comment (Sgitario) [inline 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java:155]: This is a workaround as I would have expected that deleting an application should have deleted also the services... but this is not working (I tried with all options for Cascade).
@rsvoboda do you think this is an issue worthy to be reported?

File: 011-quarkus-panache-rest-flyway/src/main/java/io/quarkus/qe/ServiceResource.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package io.quarkus.qe;
+
+import io.quarkus.hibernate.orm.rest.data.panache.PanacheEntityResource;
+
+public interface ServiceResource extends PanacheEntityResource<ServiceEntity, Long> {
```
Comment (Sgitario) [inline 011-quarkus-panache-rest-flyway/src/main/java/io/quarkus/qe/ServiceResource.java:5]: This is another workaround as I could not add the application-service relationship using the application REST endpoint... @rsvoboda do you think this is worthy to be reported as well? I would say this is a very common functionality that many users will eventually need and try.

File: 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java (hunk 1 above)
Comment (Sgitario) [reply 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java]: Finally, I've reported an issue: https://github.com/quarkusio/quarkus/issues/13842

File: 011-quarkus-panache-rest-flyway/src/main/java/io/quarkus/qe/ServiceResource.java (hunk 2 above)
Comment (Sgitario) [reply 011-quarkus-panache-rest-flyway/src/main/java/io/quarkus/qe/ServiceResource.java]: I've deleted the Service resource and reported the issue https://github.com/quarkusio/quarkus/issues/13842

**Comment by Sgitario** [comment]
Update: I've clarified how to use filters for associations. 
Still the issue between `@OneToMany` and `@JoinTable` annotations is still relevant.

File: 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java (hunk 1 above)
Comment (rsvoboda) [reply 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java]: @Sgitario Please add reference to quarkusio/quarkus#13842 as comment with TODO prefix for the future reference

**Review by rsvoboda — APPROVED** [review]
Approving, still there is one small ask before merging to add reference to quarkusio/quarkus#13842 as a comment with TODO prefix for the future reference

File: 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java (hunk 1 above)
Comment (Sgitario) [reply 011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java]: In the end, I didn't need this workaround as I found an approach to filter collections without needing the `@JoinTable` annotation. 
Though I'm monitoring the issue I reported with the limitation of usage of both `@OneToMany` and `@JoinTable` annotations.

**Pull request merged**